lib/authorizer/basic.js

```diff
@@ -42,7 +43,9 @@ module.exports = {
      * @param {Request} request
      */
     sign: function (params, request) {
-        if (!params || !params.username) { return request; } // Nothing to do if no parameters are present.
+        if (!(params && _.isString(params.username) && _.isString(params.password))) {
```

Comment:
down below in line 53, `btoa(params.username + ':' + params.password)` should change to:

```
btoa((params.username || '') + ':' + (params.password || ''))
```

Otherwise `undefined` as a string will become the username or password!!

Reply:
^ @kunagpal